src/type_conversion/jsproxy.c

```diff
@@ -363,15 +363,69 @@ JsProxy_HasBytes(PyObject* o)
   }
 }
 
-static PyObject*
-JsProxy_Dir(PyObject* o)
-{
-  JsProxy* self = (JsProxy*)o;
+#define QUIT_IF_NULL(x)                                                        \
+  do {                                                                         \
+    if (x == NULL) {                                                           \
+      goto finally;                                                            \
+    }                                                                          \
+  } while (0)
 
-  JsRef iddir = hiwire_dir(self->js);
-  PyObject* pydir = js2python(iddir);
+#define QUIT_IF_NZ(x)                                                          \
+  do {                                                                         \
+    if (x != 0) {                                                              \
+      goto finally;                                                            \
+    }                                                                          \
+  } while (0)
```

Comment:
> Would it make sense to put it into some shared header files?

Yes, I am planning to make a standard for error and refcount handling and work through the codebase one function at a time applying it everywhere. I haven't quite completed working out the details though. Can we leave this until a future PR dedicated to error handling?

> maybe `QUIT_IF_NON_ZERO`

Or something referring to the fact that I'm using it to propogate errors out? There are at least four ways to error out: allow exception to propagate as is, set a new exception eliminating the original one, set a new exception with `__cause__` the original one,. and set a new exception with `__context__` the original one. I'm not sure if we need all four of these, but I could see having four macros for that. Also `QUIT_IF_NON_ZERO` should probably be `QUIT_IF_MINUS_ONE` then it could also be used for functions that use the return value for more than a status code.

Reply:
Anyways I'm marking this resolved here, I'm planning to return to this in another PR that I may make today or tomorrow.